sdk/python/lib/pulumi/asset.py

```diff
@@ -28,51 +31,49 @@ class entity.
 
 @known_types.file_asset
 class FileAsset(Asset):
+    path: str
+
     """
     A FileAsset is a kind of asset produced from a given path to a file on
     the local filesysetm.
     """
-    def __init__(self, path):
-        # type: (str) -> None
-        if not isinstance(path, six.string_types):
+    def __init__(self, path: str) -> None:
+        if not isinstance(path, str):
```

Comment:
Just for my own education, since you were saying yesterday there's some runtime footprint for type annotations, I assume this checking is still strictly required even with the type annotations because someone could subvert/ignore them?

Reply:
Yeah, exactly.